config/confighttp/compression.go

```diff
@@ -156,6 +163,12 @@ func (d *decompressor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	if newBody != nil {
+		defer func(orig io.ReadCloser) {
+			// Ensure the original body is correctly consumed and closed
+			_, _ = io.Copy(io.Discard, orig)
+			_ = orig.Close()
+		}(r.Body)
+
```

Comment:
Is this related?

Reply:
Sorry, that is un related and needed to clean up.